### PR TITLE
Provide ability to create Hashing Strategy Sets, Maps and Bags using …

### LINKED ISSUE #1008: Provide ability to create Hashing Strategy Sets, Maps and Bags using Function
Current: 
`HashingStrategySets.mutable.with(HashingStrategies.fromFunction(Person::getName));`

The below should work once this issue is fixed:
`HashingStrategySets.mutable.with(Person::getName);`

### PR DESCRIPTION
**Pull request by goldbal330**
…Function. Closes #1008.

Signed-off-by: Alexander Goldberg <alexander.goldberg@bnymellon.com>

### CODE REVIEW

File: eclipse-collections/src/main/java/org/eclipse/collections/impl/bag/strategy/mutable/MutableHashingStrategyBagFactoryImpl.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021 Bhavana Hindupur.
+ * Copyright (c) 2022 Goldman Sachs and others.
```
Comment (donraab) [inline eclipse-collections/src/main/java/org/eclipse/collections/impl/bag/strategy/mutable/MutableHashingStrategyBagFactoryImpl.java:2]: This should be "2022 Bhavana Hindupur and others."

File: eclipse-collections-api/src/main/java/org/eclipse/collections/api/factory/bag/strategy/MutableHashingStrategyBagFactory.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021 Bhavana Hindupur.
+ * Copyright (c) 2022 Goldman Sachs and others.
```
Comment (donraab) [inline eclipse-collections-api/src/main/java/org/eclipse/collections/api/factory/bag/strategy/MutableHashingStrategyBagFactory.java:2]: This should be "2022 Bhavana Hindupur and others."

File: unit-tests/src/test/java/org/eclipse/collections/impl/factory/HashingStrategyBagsTest.java [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021 Bhavana Hindupur.
+ * Copyright (c) 2022 Goldman Sachs and others.
```
Comment (donraab) [inline unit-tests/src/test/java/org/eclipse/collections/impl/factory/HashingStrategyBagsTest.java:2]: This should be "2022 Bhavana Hindupur and others."

**Review by donraab — CHANGES_REQUESTED** [review]
A few minor changes requested.

File: eclipse-collections/src/main/java/org/eclipse/collections/impl/bag/strategy/mutable/MutableHashingStrategyBagFactoryImpl.java (hunk 1 above)
Comment (goldbal330) [reply eclipse-collections/src/main/java/org/eclipse/collections/impl/bag/strategy/mutable/MutableHashingStrategyBagFactoryImpl.java]: addressed

File: eclipse-collections-api/src/main/java/org/eclipse/collections/api/factory/bag/strategy/MutableHashingStrategyBagFactory.java (hunk 2 above)
Comment (goldbal330) [reply eclipse-collections-api/src/main/java/org/eclipse/collections/api/factory/bag/strategy/MutableHashingStrategyBagFactory.java]: done

File: unit-tests/src/test/java/org/eclipse/collections/impl/factory/HashingStrategyBagsTest.java (hunk 3 above)
Comment (goldbal330) [reply unit-tests/src/test/java/org/eclipse/collections/impl/factory/HashingStrategyBagsTest.java]: fixed

**Review by donraab — APPROVED** [review]

**Pull request merged**